tests/unit/test_coordinate_utilities.py

```diff
@@ -0,0 +1,515 @@
+
+        expected_override_dimensions_AM = [
+            '/Soil_Moisture_Retrieval_Data_AM/projected_y',
+            '/Soil_Moisture_Retrieval_Data_AM/projected_x',
+        ]
+        expected_override_dimensions_PM = [
+            '/Soil_Moisture_Retrieval_Data_PM/projected_y',
+            '/Soil_Moisture_Retrieval_Data_PM/projected_x',
+        ]
+
+        with self.subTest(
+            'Retrieves expected override dimensions for the science variable'
+        ):
+            self.assertListEqual(
+                get_projected_dimension_names_from_coordinate_variables(
+                    self.varinfo, '/Soil_Moisture_Retrieval_Data_AM/surface_flag'
+                ),
+                expected_override_dimensions_AM,
+            )
+
+        with self.subTest(
+            'Retrieves expected override dimensions for the longitude variable'
+        ):
+            self.assertListEqual(
+                get_projected_dimension_names_from_coordinate_variables(
+                    self.varinfo, self.longitude
+                ),
+                expected_override_dimensions_AM,
+            )
+
+        with self.subTest(
+            'Retrieves expected override dimensions for the latitude variable'
+        ):
+            self.assertListEqual(
+                get_projected_dimension_names_from_coordinate_variables(
+                    self.varinfo, self.latitude
+                ),
+                expected_override_dimensions_AM,
+            )
+
+        with self.subTest(
+            'Retrieves expected override dimensions science variable with a different grid'
+        ):
+            self.assertListEqual(
+                get_projected_dimension_names_from_coordinate_variables(
+                    self.varinfo, '/Soil_Moisture_Retrieval_Data_PM/surface_flag_pm'
+                ),
+                expected_override_dimensions_PM,
+            )
+        with self.subTest(
+            'Retrieves empty dimensions list when science variable has no coordinates'
+        ):
+            self.assertListEqual(
+                get_projected_dimension_names_from_coordinate_variables(
+                    self.varinfo, '/Soil_Moisture_Retrieval_Data_PM/surface_flag_pm'
+                ),
+                expected_override_dimensions_PM,
+            )
+
+    def test_get_row_col_sizes_from_coordinate_datasets(self):
+        """Ensure that the correct row and column sizes are
+        returned for the requested coordinates
+        """
+        expected_row_col_sizes = (5, 10)
+        lat_1d_array = np.array([1, 2, 3, 4])
+        lon_1d_array = np.array([6, 7, 8, 9])
+        lat_mismatched_array = np.array([[1, 2, 3], [3, 4, 5]])
+        lon_mismatched_array = np.array([[6, 7], [8, 9], [10, 11]])
+        lat_empty_ndim_array = np.array(
+            0,
+        )
+        lon_empty_ndim_array = np.array(
+            0,
+        )
+        lat_empty_size_array = np.array([])
+        lon_empty_size_array = np.array([])
```

Comment:
I think I mentioned this somewhere else, but it's a lot easier for the reader if these things that relate to specific subtests were within the relevant `with self.subTest(...)` context manager. From a quick skim, it doesn't look like these items are used in multiple subtests, so moving them shouldn't break anything. (Note - subtests aren't always executed in the order that they are written - I think `unittest` orders things by alphabetising the test classes, methods and subtest descriptions in some way)

It's not a big deal, and not something to hold a PR up over, but it's a nice to have, which means I can look at a single subtest and see what's relevant to it in a single look.

Reply:
on the earlier comment on get_fill_value function  - 
If I don't do the (float) typecast when I get the fill_value, I would get the following error. because the get_attribute _value method in varinfo returns a str value
 
TypeError: The DType <class 'numpy._FloatAbstractDType'> could not be promoted by <class 'numpy.dtype[str_]'>.

Reply:
updated the get_valid_indices function. Had to add [0] since np.where returns a tuple